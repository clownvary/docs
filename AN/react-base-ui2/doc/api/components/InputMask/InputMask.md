## InputMask  extends InputBase



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---