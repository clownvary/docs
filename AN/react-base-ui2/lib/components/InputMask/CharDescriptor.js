'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var CharDescriptor =
function CharDescriptor() {var maskPosition = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;var charType = arguments[1];(0, _classCallCheck3.default)(this, CharDescriptor);
  this.maskPosition = maskPosition;
  this.charType = charType;
  this.isAssigned = false;
  this.caseType = '';
};exports.default =


CharDescriptor;module.exports = exports['default'];