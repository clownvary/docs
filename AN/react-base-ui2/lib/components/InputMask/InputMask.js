'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.InputMask = undefined;var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _get2 = require('babel-runtime/helpers/get');var _get3 = _interopRequireDefault(_get2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _InputBase2 = require('../InputBase');var _InputBase3 = _interopRequireDefault(_InputBase2);
var _MaskTextProvider = require('./MaskTextProvider');var _MaskTextProvider2 = _interopRequireDefault(_MaskTextProvider);
var _consts = require('../../consts');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}


/**
                                                                                                                                     * Default Props for InputMask
                                                                                                                                     * @name InputMaskProps
                                                                                                                                     * @const
                                                                                                                                     */
var InputMaskProps = {
  /**
                        * Determines the input mask to use at run time.
                        * Mask must be a string composed of one or more of the masking elements.
                        *
                        * The supported mask characters are:
                        *
                        * - 0= Digit
                        * - 9= Digit or space
                        * - #= Digit, sign, or space
                        * - L= Letter
                        * - l= Letter or space
                        * - A= Alphanumeric
                        * - a= Alphanumeric or space
                        * - .= Localized decimal point
                        * - ,= Localized thousand separator
                        * - := Localized time separator
                        * - /= Localized date separator
                        * - $= Localized currency symbol
                        * - <= Converts characters that follow to lowercase
                        * - \>= Converts characters that follow to uppercase
                        * - |= Disables case conversion
                        * - \= scapes any character, turning it into a literal
                        *
                        */
  mask: '',
  /**
             * Determines the character used to represent the absence of user input.
             */
  promptChar: '_',
  /**
                    * Determines the character to be substituted for the actual input characters in password mode.
                    */
  passwordChar: '*',
  /**
                      * Determines whether password char is used.
                      */
  passwordMode: false,
  /**
                        * Indicates whether the prompt characters in the input mask are hidden
                        * when the input loses focus.
                        */
  hidePromptOnLeave: false,
  /**
                             * Indicates whether promptChar can be entered as valid data by the user.
                             */
  allowPromptAsInput: false };


var defaultProps = (0, _extends3.default)({}, _InputBase2.defaultProps, InputMaskProps);

/**
                                                                                          * @description
                                                                                          * InputMask component allows you to validate and format user input
                                                                                          * while typing, this will prevent invalid data been input.
                                                                                          *
                                                                                          * To use the InputMask component, set the mask property to a string that specifies
                                                                                          * the valid character classes for each field.
                                                                                          *
                                                                                          * @class
                                                                                          */var
InputMask = exports.InputMask = function (_InputBase) {(0, _inherits3.default)(InputMask, _InputBase);function InputMask() {(0, _classCallCheck3.default)(this, InputMask);return (0, _possibleConstructorReturn3.default)(this, (InputMask.__proto__ || (0, _getPrototypeOf2.default)(InputMask)).apply(this, arguments));}(0, _createClass3.default)(InputMask, [{ key: 'createTextProvider', value: function createTextProvider()











    {
      return new _MaskTextProvider2.default(this.props);
    } /**
       * Default Props of InputMask.
       *
       * Please see InputMaskProps for details.
       *
       * @see InputMaskProps
       */ }, { key: 'getProviderOptions', value: function getProviderOptions() {var _props = this.props,hidePromptOnLeave = _props.hidePromptOnLeave,passwordMode = _props.passwordMode;var includePrompt = hidePromptOnLeave ? this.isFocused() : true;var includeLiterals = true;return {
        includePrompt: includePrompt,
        includeLiterals: includeLiterals,
        passwordMode: passwordMode };

    } }, { key: 'getContainerClassName', value: function getContainerClassName()

    {
      return _consts.DefaultCSSPrefix + '-input-mask';
    } }, { key: 'onInputFocus', value: function onInputFocus(

    e) {
      if (this.textProvider && this.allowEdit()) {var
        hidePromptOnLeave = this.props.hidePromptOnLeave;
        if (hidePromptOnLeave) {
          this.updateText({ start: 0, end: 0 });
        }
      }

      (0, _get3.default)(InputMask.prototype.__proto__ || (0, _getPrototypeOf2.default)(InputMask.prototype), 'onInputFocus', this).call(this, e);
    } }, { key: 'onInputBlur', value: function onInputBlur(

    e) {var _this2 = this;
      if (this.textProvider && this.allowEdit()) {var
        hidePromptOnLeave = this.props.hidePromptOnLeave;
        if (hidePromptOnLeave) {
          setTimeout(function () {return _this2.updateText();}, 1);
        }
      }

      (0, _get3.default)(InputMask.prototype.__proto__ || (0, _getPrototypeOf2.default)(InputMask.prototype), 'onInputBlur', this).call(this, e);
    } }, { key: 'getTextWithPrompts', value: function getTextWithPrompts()

    {
      return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: true,
        includeLiterals: false });

    } }, { key: 'getTextWithLiterals', value: function getTextWithLiterals()

    {
      return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: false,
        includeLiterals: true });

    } }, { key: 'getTextWithPromptAndLiterals', value: function getTextWithPromptAndLiterals()

    {
      return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: true,
        includeLiterals: true });

    } }]);return InputMask;}(_InputBase3.default);


/**
                                                    * @react-component
                                                    */InputMask.displayName = 'InputMask';InputMask.defaultProps = defaultProps;exports.default =
InputMask;