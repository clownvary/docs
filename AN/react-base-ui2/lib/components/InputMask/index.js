'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _InputMask = require('./InputMask');var _InputMask2 = _interopRequireDefault(_InputMask);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

_InputMask2.default;module.exports = exports['default'];