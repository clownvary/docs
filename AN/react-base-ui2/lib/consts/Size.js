'use strict';Object.defineProperty(exports, "__esModule", { value: true }); /** General Size Enum.
                                                                             * @enum {string}
                                                                            */
var Size = {
  EXTRA_SMALL: 'xs',
  SMALL: 'sm',
  MEDIUM: 'md',
  LARGE: 'lg',
  EXTRA_LARGE: 'xl' };


var Size3 = {
  SMALL: 'sm',
  MEDIUM: 'md',
  LARGE: 'lg' };


var Size5 = Size;exports.default =

Size;exports.


Size = Size;exports.
Size3 = Size3;exports.
Size5 = Size5;