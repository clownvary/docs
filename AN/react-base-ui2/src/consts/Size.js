/** General Size Enum.
 * @enum {string}
*/
const Size = {
  EXTRA_SMALL: 'xs',
  SMALL: 'sm',
  MEDIUM: 'md',
  LARGE: 'lg',
  EXTRA_LARGE: 'xl'
};

const Size3 = {
  SMALL: 'sm',
  MEDIUM: 'md',
  LARGE: 'lg'
};

const Size5 = Size;

export default Size;

export {
  Size,
  Size3,
  Size5
};
