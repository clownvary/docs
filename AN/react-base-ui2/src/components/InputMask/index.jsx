import InputMask from './InputMask';

export default InputMask;
