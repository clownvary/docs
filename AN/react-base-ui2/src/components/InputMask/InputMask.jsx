import InputBase, { defaultProps as baseProps } from '../InputBase';
import MaskTextProvider from './MaskTextProvider';
import { DefaultCSSPrefix } from '../../consts';


/**
 * Default Props for InputMask
 * @name InputMaskProps
 * @const
 */
const InputMaskProps = {
  /**
   * Determines the input mask to use at run time.
   * Mask must be a string composed of one or more of the masking elements.
   *
   * The supported mask characters are:
   *
   * - 0= Digit
   * - 9= Digit or space
   * - #= Digit, sign, or space
   * - L= Letter
   * - l= Letter or space
   * - A= Alphanumeric
   * - a= Alphanumeric or space
   * - .= Localized decimal point
   * - ,= Localized thousand separator
   * - := Localized time separator
   * - /= Localized date separator
   * - $= Localized currency symbol
   * - <= Converts characters that follow to lowercase
   * - \>= Converts characters that follow to uppercase
   * - |= Disables case conversion
   * - \= scapes any character, turning it into a literal
   *
   */
  mask: '',
  /**
   * Determines the character used to represent the absence of user input.
   */
  promptChar: '_',
  /**
   * Determines the character to be substituted for the actual input characters in password mode.
   */
  passwordChar: '*',
  /**
   * Determines whether password char is used.
   */
  passwordMode: false,
  /**
   * Indicates whether the prompt characters in the input mask are hidden
   * when the input loses focus.
   */
  hidePromptOnLeave: false,
  /**
   * Indicates whether promptChar can be entered as valid data by the user.
   */
  allowPromptAsInput: false
};

const defaultProps = { ...baseProps, ...InputMaskProps };

/**
 * @description
 * InputMask component allows you to validate and format user input
 * while typing, this will prevent invalid data been input.
 *
 * To use the InputMask component, set the mask property to a string that specifies
 * the valid character classes for each field.
 *
 * @class
 */
export class InputMask extends InputBase {

  static displayName = 'InputMask';
  /**
   * Default Props of InputMask.
   *
   * Please see InputMaskProps for details.
   *
   * @see InputMaskProps
   */
  static defaultProps = defaultProps;

  createTextProvider() {
    return new MaskTextProvider(this.props);
  }

  getProviderOptions() {
    const { hidePromptOnLeave, passwordMode } = this.props;
    const includePrompt = hidePromptOnLeave ? this.isFocused() : true;
    const includeLiterals = true;
    return {
      includePrompt,
      includeLiterals,
      passwordMode
    };
  }

  getContainerClassName() {
    return `${DefaultCSSPrefix}-input-mask`;
  }

  onInputFocus(e) {
    if (this.textProvider && this.allowEdit()) {
      const { hidePromptOnLeave } = this.props;
      if (hidePromptOnLeave) {
        this.updateText({ start: 0, end: 0 });
      }
    }

    super.onInputFocus(e);
  }

  onInputBlur(e) {
    if (this.textProvider && this.allowEdit()) {
      const { hidePromptOnLeave } = this.props;
      if (hidePromptOnLeave) {
        setTimeout(() => this.updateText(), 1);
      }
    }

    super.onInputBlur(e);
  }

  getTextWithPrompts() {
    return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: true,
        includeLiterals: false
      });
  }

  getTextWithLiterals() {
    return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: false,
        includeLiterals: true
      });
  }

  getTextWithPromptAndLiterals() {
    return !this.textProvider ? this.input.value :
      this.textProvider.getText({
        includePrompt: true,
        includeLiterals: true
      });
  }
}

/**
 * @react-component
 */
export default InputMask;
