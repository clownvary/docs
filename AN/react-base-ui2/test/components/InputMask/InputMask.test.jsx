import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import InputMask from 'src/components/InputMask';
import Size from 'src/consts/Size';

describe('components/InputMask/InputMask.jsx', () => {
  test('basic usage', async () => {
    const wrapper = mount(<InputMask size={Size.SMALL} />);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const input = wrapper.find('input');
    input.simulate('focus');
    expect(toJSON(wrapper)).toMatchSnapshot();

    input.simulate('blur');
    expect(toJSON(wrapper)).toMatchSnapshot();

    const instance = wrapper.instance();
    expect(instance.getTextWithPrompts()).toBe('');
    expect(instance.getTextWithLiterals()).toBe('');
    expect(instance.getTextWithPromptAndLiterals()).toBe('');

    wrapper.setProps({ hidePromptOnLeave: true });
    instance.updateText = jest.fn();

    input.simulate('focus');
    expect(instance.updateText).toBeCalledWith({ start: 0, end: 0 });

    await input.simulate('blur');
    expect(instance.updateText).toBeCalled();

    instance.updateText.mockClear();
    delete instance.textProvider;
    input.simulate('focus');
    expect(instance.updateText).not.toBeCalled();
    await input.simulate('blur');
    expect(instance.updateText).not.toBeCalled();

    expect(instance.getTextWithPrompts()).toBe('');
    expect(instance.getTextWithLiterals()).toBe('');
    expect(instance.getTextWithPromptAndLiterals()).toBe('');
  });
});
