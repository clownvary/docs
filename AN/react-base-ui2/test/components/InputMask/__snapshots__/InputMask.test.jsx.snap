// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InputMask/InputMask.jsx basic usage 1`] = `
<InputMask
  allowPromptAsInput={false}
  hidePromptOnLeave={false}
  mask=""
  passwordChar="*"
  passwordMode={false}
  promptChar="_"
  size="sm"
>
  <div
    className="an-input-mask input-group buttons-0 input-group--sm"
  >
    <input
      className="input input-group__field textalign-left input__field"
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      type="text"
    />
  </div>
</InputMask>
`;

exports[`components/InputMask/InputMask.jsx basic usage 2`] = `
<InputMask
  allowPromptAsInput={false}
  hidePromptOnLeave={false}
  mask=""
  passwordChar="*"
  passwordMode={false}
  promptChar="_"
  size="sm"
>
  <div
    className="an-input-mask input-group buttons-0 input-group--sm"
  >
    <input
      className="input input-group__field textalign-left input__field"
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      type="text"
    />
  </div>
</InputMask>
`;

exports[`components/InputMask/InputMask.jsx basic usage 3`] = `
<InputMask
  allowPromptAsInput={false}
  hidePromptOnLeave={false}
  mask=""
  passwordChar="*"
  passwordMode={false}
  promptChar="_"
  size="sm"
>
  <div
    className="an-input-mask input-group buttons-0 input-group--sm"
  >
    <input
      className="input input-group__field textalign-left input__field"
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      type="text"
    />
  </div>
</InputMask>
`;
